src/zygoterules.jl

```diff
@@ -1,17 +1,35 @@
-using ZygoteRules
+# # This is a performance hack specifically for Zygote, because it doesn't handle fused
+# # broadcasts well
+# for (f, df) in [
+#     (:relu, :(x .> 0)),
+#     (:selu, :(dselu.(x))),
+#     (:elu, :(delu.(x))),
+#     (:σ, :(conj.(Ω .* (1 .- Ω)))),
+# ]
+#     pullback = Symbol(:broadcasted_, f, :_pullback)
+#     @eval @adjoint function Base.Broadcast.broadcasted(::typeof($f), x::Numeric)
```

Comment:
I am not sure this is a good idea, since IIUC, loading NNlib will now always error if Zygote is not loaded first. Perhaps we can use Requires.jl for this instead?

Reply:
I don't understand, when should it error? I can do `using NNlib`  just fine

Reply:
Using requires is not an option I think, because Zygote v0.5 precompilation triggers NNlib's one, NNlib define the rules, then Zygote tries to define the same rules and gives the warning

Reply:
> I don't understand, when should it error? I can do `using NNlib`  just fine

I can't try it ATM, but since it doesn't load `ZygoteRules` now, I don't know where it should get the definition of `@adjoint` from. It might just work for the tests, since those already load Zygote. Did you do `using NNlib` before loading any other packages?

Reply:
> Using requires is not an option I think, because Zygote v0.5 precompilation triggers NNlib's one, NNlib define the rules, then Zygote tries to define the same rules and gives the warning

I think we could use requires for checking if Zygote is loaded and then only define those rules if it's the appropriate version.

Reply:
> I don't know where it should get the definition of @adjoint from

ah, maybe you didn't see the new rules down below?

> I think we could use requires for checking if Zygote is loaded and then only define those rules if it's the appropriate version.

I think this won't work because Zygote v0.5 depends on NNlib, so NNlib is always loaded first

Reply:
> I don't know where it should get the definition of `@adjoint` from

maybe there is been a misunderstanding, I replaced ZygoteRules' adjoint with ChainRules' rrules. I don't know why it wasn't done before, maybe because is not something any AD engine would want to do

Reply:
🤦I didn't see this was commented out for some reason, sorry! If this is just a temporary workaround, using ChainRulesCore for these seems ok to me, but we really want to avoid this once people are actually using other ADs than Zygote. https://github.com/JuliaDiff/ChainRulesCore.jl/issues/270 might be helpful here.